libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -749,140 +953,176 @@ uint16_t GCS::sys_status_errors1()
   handle an incoming mission item
   return true if this is the last mission item, otherwise false
  */
-bool GCS_MAVLINK::handle_mission_item(mavlink_message_t *msg, AP_Mission &mission)
+void GCS_MAVLINK::handle_mission_item(const mavlink_message_t *msg)
 {
-    MAV_MISSION_RESULT result = MAV_MISSION_ACCEPTED;
-    struct AP_Mission::Mission_Command cmd = {};
-    bool mission_is_complete = false;
-    uint16_t seq=0;
-    uint16_t current = 0;
-    
+    // TODO: rename packet to mission_item_int
+    mavlink_mission_item_int_t packet;
     if (msg->msgid == MAVLINK_MSG_ID_MISSION_ITEM) {      
-        mavlink_mission_item_t packet;    
-        mavlink_msg_mission_item_decode(msg, &packet);
-        
-        // convert mavlink packet to mission command
-        result = AP_Mission::mavlink_to_mission_cmd(packet, cmd);
-        if (result != MAV_MISSION_ACCEPTED) {
-            goto mission_ack;
+        mavlink_mission_item_t mission_item;
+        mavlink_msg_mission_item_decode(msg, &mission_item);
+        MAV_MISSION_RESULT ret = AP_Mission::convert_MISSION_ITEM_to_MISSION_ITEM_INT(mission_item, packet);
+        if (ret != MAV_MISSION_ACCEPTED) {
+            const MAV_MISSION_TYPE type = (MAV_MISSION_TYPE)packet.mission_type;
+            send_mission_ack(*msg, type, ret);
+            return;
         }
-        
-        seq = packet.seq;
-        current = packet.current;
     } else {
-        mavlink_mission_item_int_t packet;
         mavlink_msg_mission_item_int_decode(msg, &packet);
-        
-        // convert mavlink packet to mission command
-        result = AP_Mission::mavlink_int_to_mission_cmd(packet, cmd);
+    }
+    const int16_t current = packet.current;
+    const MAV_MISSION_TYPE type = (MAV_MISSION_TYPE)packet.mission_type;
```

Comment:
You can do this earlier in the function, you've declared the symbol twice now in different scopes, might as well just do it early once.

Reply:
Doing it this way allows us to `const` it and doesn't cost anything.

Reply:
Yeah, I see that now, sorry!